datafusion/physical-optimizer/Cargo.toml

```diff
@@ -33,13 +33,17 @@ workspace = true
 
 [dependencies]
 arrow = { workspace = true }
+arrow-array = { workspace = true }
 arrow-schema = { workspace = true }
 datafusion-common = { workspace = true, default-features = true }
 datafusion-execution = { workspace = true }
 datafusion-expr-common = { workspace = true, default-features = true }
+datafusion-expr = { workspace = true, default-features = true }
+datafusion-functions-nested = { workspace = true }
```

Comment:
Can you please move `datafusion-functions-nested` to `dev-dependencies`?

I also think we can avoid the dependency on `datafusion-expr` by changing 

```rust
use datafusion_expr::Operator;
```

to 

```rust
use datafusion_expr_common::operator::Operator;
```